js/util/window.js

```diff
@@ -30,12 +30,19 @@ function restore() {
     window.HTMLElement.prototype.clientLeft = 0;
     window.HTMLElement.prototype.clientTop = 0;
 
-    window.HTMLCanvasElement.prototype.getContext = function(type, attributes) {
-        if (!this._webGLContext) {
-            this._webGLContext = gl(this.width, this.height, attributes);
-        }
-        return this._webGLContext;
-    };
+    window.HTMLCanvasElement.prototype.getContext = (function () {
+        let original = window.HTMLCanvasElement.prototype.getContext;
+        return function (type, attributes) {
+            if (type === 'webgl') {
+                if (!this._webGLContext) {
+                    this._webGLContext = gl(this.width, this.height, attributes);
+                }
+                return this._webGLContext;
+            }
+            original = original.bind(this);
+            return original(type, attributes);
```

Comment:
Instead of doing bind and then calling the function, you could just do `original.call(this, type, attributes);`.